src/modules/ims_usrloc_pcscf/ul_callback.c

```diff
@@ -98,6 +98,39 @@ void destroy_ul_callbacks_list(struct ul_callback* cb) {
 	}
 }
 
+int register_ulcb_method( struct pcontact *c, int types, ul_cb f, void *param )
+{
+        //struct ul_callback *cbp;
+
+        /* are the callback types valid?... */
+        if ( types<0 || types>PCSCF_MAX ) {
+                LM_CRIT("invalid callback types: mask=%d\n",types);
+                return E_BUG;
+        }
+        /* we don't register null functions */
+        if (f==0) {
+                LM_CRIT("null callback function\n");
+                return E_BUG;
+        }
+
+        /* build a new callback structure */
+        if ( types & PCSCF_CONTACT_UPDATE){
+            if (!(cbp_registrar=(struct ul_callback*)shm_malloc(sizeof( struct ul_callback)))) {
+                LM_ERR("no more share mem\n");
+                return E_OUT_OF_MEM;
+            }
+            cbp_registrar->callback = f;
+        }
+        else{
+            if (!(cbp_qos=(struct ul_callback*)shm_malloc(sizeof( struct ul_callback)))) {
```

Comment:
This variable is causing a conflict with ims_qos module:
/usr/bin/ld: ims_qos_mod.o:./src/modules/ims_qos/./../ims_usrloc_pcscf/../ims_usrloc_pcscf/usrloc.h:246: multiple definition of `cbp_qos'; cdpeventprocessor.o:./src/modules/ims_qos/./../ims_usrloc_pcscf/../ims_usrloc_pcscf/usrloc.h:246: first defined here

Should be renamed